template/script.sh.erb

```diff
@@ -1,38 +1,57 @@
 #!/usr/bin/env bash
-<%-
-    # Ensure that code-server always starts up in either a user defined directory or the home directory
-    working_dir = Pathname.new(context.working_dir)
-    if ! working_dir.exist? || working_dir.to_s.empty?
-        working_dir = Pathname.new(ENV['HOME'])
-    elsif working_dir.file?
-        working_dir = working_dir.parent
-    end
+<%
+# Determine the application root from session token.
+app_root = BatchConnect::App.from_token(session.token).root.realpath
+
+# Set our working directory.
+working_dir = Pathname.new(context.working_dir)
```

Comment:
Is coerceing to string necessary or is working_dir guarenteed to never be nil? `working_dir = Pathname.new(context.working_dir.to_s)`